test/correctness/respect_input_constraint_in_bounds_inference.cpp

```diff
@@ -0,0 +1,23 @@
+#include "Halide.h"
+
+using namespace Halide;
+
+int main(int argc, char **argv) {
+    ImageParam im(Float(32), 1);
+    Func f, g;
+    Var x;
+
+    f(x) = x;
+    g(x) = f(x % im.dim(0).extent());
+    im.dim(0).set_extent(16);
+
+    // Given the constraint, we know the bounds of f should be less than 16, so
```

Comment:
Suggestion: add a variant where bounds of f is >= (say) 1024 and make it an error test?